src/main/java/org/apache/ibatis/type/DateOnlyTypeHandler.java

```diff
@@ -33,21 +33,21 @@ public void setNonNullParameter(PreparedStatement ps, int i, Date parameter, Jdb
 
   @Override
   public Date getNullableResult(ResultSet rs, String columnName) throws SQLException {
-    return toSqlDate(rs.getDate(columnName));
+    return toDate(rs.getDate(columnName));
   }
 
   @Override
   public Date getNullableResult(ResultSet rs, int columnIndex) throws SQLException {
-    return toSqlDate(rs.getDate(columnIndex));
+    return toDate(rs.getDate(columnIndex));
   }
 
   @Override
   public Date getNullableResult(CallableStatement cs, int columnIndex) throws SQLException {
-    return toSqlDate(cs.getDate(columnIndex));
+    return toDate(cs.getDate(columnIndex));
   }
 
-  private java.sql.Date toSqlDate(Date date) {
-    return date == null ? null : new java.sql.Date(date.getTime());
+  private Date toDate(Date date) {
```

Comment:
Please change the argument type to `java.sql.Date` (to make the purpose of the method clearer).